lib/repo_tools.pm

```diff
@@ -147,11 +147,19 @@ sub prepare_source_repo {
     }
     # source repository is disabled by default
     else {
+        # OSS_SOURCE is expected to be added
         if (script_run('zypper lr repo-source') != 0) {
             # re-add the source repo
             my $version = lc get_required_var('VERSION');
-            my $source_name = is_tumbleweed() ? $version : 'distribution/leap/' . $version;
-            zypper_call("ar -f http://download.opensuse.org/source/$source_name/repo/oss repo-source");
+            my $repourl;
+            # if REPO_OSS_SOURCE is defined - use it, if not fallback to download.opensuse.org
+            if (my $repo_basename = get_var("REPO_OSS_SOURCE")) {
+                $repourl = get_required_var('MIRROR_PREFIX') . "/" . $repo_basename;
```

Comment:
if I correctly get idea to use this flow I need to define two variables,  containing parts of url , what the benefit in comparison just put url into single variable ?

Reply:
It's more about backward compatibility and what we already had, `MIRROR_PREFIX` is defined by rsync.pl since I know it, and for `REPO_` variables we have never used full url in a single variable. In general that could work and provide a better way to provide repos from different hosts, but that's the way it is and it will be a lot of efforts for not much benefit (as of now) to change it.